src/System.Net.Primitives/tests/FunctionalTests/IPEndPointParsing.cs

```diff
@@ -0,0 +1,305 @@
+        public void Parse_ValidEndPoint_IPv6_Success(string address, string expectedAddress)
+        {
+            Parse_ValidEndPoint_Success(address, expectedAddress, false);
+        }
+
+        private void Parse_ValidEndPoint_Success(string address, string expectedAddress, bool isIPv4)
+        {
+            // We'll parse just the address alone followed by the address with various port numbers
+
+            expectedAddress = expectedAddress.ToLowerInvariant();   // This is done in the IP parse routines
+
+            // TryParse should return true
+            Assert.True(IPEndPoint.TryParse(address, out IPEndPoint result));
+            Assert.Equal(expectedAddress, result.Address.ToString());
+            Assert.Equal(0, result.Port);
+
+            // Parse should give us the same result
+            result = IPEndPoint.Parse(address);
+            Assert.Equal(expectedAddress, result.Address.ToString());
+            Assert.Equal(0, result.Port);
+
+            // Cover varying lengths of port number
+            int portNumber = 1;
+            for (int i = 0; i < 5; i++)
+            {
+                var addressAndPort = isIPv4 ? $"{address}:{i}" : $"[{address}]:{i}";
+
+                // TryParse should return true
+                Assert.True(IPEndPoint.TryParse(addressAndPort, out result));
+                Assert.Equal(expectedAddress, result.Address.ToString());
+                Assert.Equal(i, result.Port);
+
+                // Parse should give us the same result
+                result = IPEndPoint.Parse(addressAndPort);
+                Assert.Equal(expectedAddress, result.Address.ToString());
+                Assert.Equal(i, result.Port);
+
+                // i.e.: 1; 12; 123; 1234; 12345
+                portNumber *= 10;
+                portNumber += i + 2;
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(IPAddressParsing.InvalidIpv4Addresses), MemberType = typeof(IPAddressParsing))]
+        [MemberData(nameof(IPAddressParsing.InvalidIpv4AddressesStandalone), MemberType = typeof(IPAddressParsing))]
+        public void Parse_InvalidAddress_IPv4_Throws(string address)
+        {
+            Parse_InvalidAddress_Throws(address, true);            
+        }
+
+        [Theory]
+        [MemberData(nameof(IPAddressParsing.InvalidIpv6Addresses), MemberType = typeof(IPAddressParsing))]
+        [MemberData(nameof(IPAddressParsing.InvalidIpv6AddressesNoInner), MemberType = typeof(IPAddressParsing))]
+        public void Parse_InvalidAddress_IPv6_Throws(string address)
+        {
+            Parse_InvalidAddress_Throws(address, false);
+        }
+
+        private void Parse_InvalidAddress_Throws(string address, bool isIPv4)
+        {
+            // TryParse should return false and set result to null
+            var result = new IPEndPoint(IPAddress.Parse("0"), 25);
```

Comment:
Shouldn't this be using `address` instead of `"0"`?

Reply:
But... why is result being initialized to a new IPEndPoint, only to immediately be overwritten by the TryParse on the next line?